### PR TITLE
PR#7007, PR#7311: document OCAMLPARAM and ocaml_compiler_internal_params

### PR DESCRIPTION
**Pull request by damiendoligez**
Here is my first draft. Comments and suggestions are welcome.

### CODE REVIEW

File: manual/manual/cmds/comp.etex [hunk 1]
```diff
@@ -552,11 +552,22 @@ Display a short usage summary and exit.
 %
 \end{options}
 
-\noindent
-On native Windows, the following environment variable is also consulted:
+\paragraph{contextual control of command-line options}
```
Comment (alainfrisch) [inline manual/manual/cmds/comp.etex]: I would use a capitalized title here ("Contextual").

Comment (mshinwell) [inline manual/manual/cmds/comp.etex]: Capital C

File: manual/manual/cmds/comp.etex [hunk 2]
```diff
@@ -552,11 +552,22 @@ Display a short usage summary and exit.
...
+
+The compiler command line can be modified ``from the outside''
+with the following mechanisms. These are experimental
+and subject to change, and they are not intended for normal use.
```
Comment (mshinwell) [inline manual/manual/cmds/comp.etex]: What about "and subject to change.  They should be used only for experimental and development work---never in released packages." ?

File: manual/manual/cmds/native.etex [hunk 3]
```diff
@@ -566,6 +558,25 @@ The Sparc code generator supports the following additional options:
...
+\item["ocaml_compiler_internal_params" \rm(file in the stdlib directory)]
+A mapping of file names to lists of arguments that
+will be added to the command line (and "OCAMLPARAM") arguments.
+\item["OCAML_FLEXLINK" \rm(environment variable)]
+Alternative executable to use on native
+Windows for "flexlink" instead of the
+configured value. Primarily used for bootstrapping.
+\end{options}
+
 \section{Common errors}
```
Comment (mshinwell) [inline manual/manual/cmds/native.etex:581]: Can we "include" this section from a separate file maybe?

**Review by alainfrisch — COMMENTED** [review]
Looks good to me.  I don't think that the information is enough to use either ocaml_compiler_internal_params or OCAMLPARAM, but the text carry the intented message ("do not use!").

**Comment by dra27** [comment]
+1 - looks good to me

**Comment by gasche** [comment]
That's a good minimal first step. As a user when reading this documentation I would have liked to see examples of the syntax, with the following questions in mind:
- for OCAMLPARAM, what is evaluated before and after command-line parsing?
- for ocaml_internal_params, the documentation says that specific file paths/names are mapped to options, but are those relative or absolute names?
- I suppose there is a wildcard syntax to add an option to all files, what is its syntax?

File: manual/manual/cmds/native.etex (hunk 3 above)
Comment (damiendoligez) [reply manual/manual/cmds/native.etex:581]: There's a lot of duplicate text between `comp.etex` and `native.etex`. Factoring it out may be a good idea, but I won't attempt it in this PR.

**Comment by damiendoligez** [comment]
@gasche The idea (for the moment) is to document their existence so that users who see their compiler behaving weirdly will have a chance to figure out what is going on. Do you agree to merge this minimal first step for 4.04, then work on expanding the documentation in trunk?

**Comment by gasche** [comment]
Yep, sure.

File: manual/manual/cmds/native.etex (hunk 3 above)
Comment (Octachron) [reply manual/manual/cmds/native.etex:581]: Concerning the duplication between `comp.etex` and `native.etex`, I had proposed to merge these files together (with `top.etex`) and handle the differences with tex macros in #491 . If there are some interests in this solution, I could rebase this PR; or create a version merging only `comp.etex` and `native.etex` − which would decrease the complexity of the resulting tex file.

**Pull request merged**